### PR TITLE
feat: challenges part 2 ex9

### PR DESCRIPTION
**Pull request by PedroPapa1**
9. **Salary and Gender Report**
   Develop an application that reads the salary and gender of several employees. At the end, show the total salary paid to men and the total paid to women. The program should ask the user if they want to continue or not whenever it reads the data of an employee.

### CODE REVIEW

File: list2/ex9/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export function salaryAndGenderReport() {
+    let continueLoop;
+    let totalMenSalary = 0;
+    let totalFemaleSalary = 0;
+
+    do {
+        console.log('What is the gender of the employee? (Insert M to Male, and F to Female)');
+        const genderPrompt = prompt('>').toUpperCase();
+
+        if (genderPrompt !== 'F' && genderPrompt !== 'M') {
```
Comment (emiliosheinz) [inline list2/ex9/index.js]: You can abstract `F` and `M` into a constant.

```js
const GENDERS = {
  FEMALE = 'F',
  MALE = 'M',
}
```

and use it here to validate the input

```js
if(!Object.values(GENDERS).includes(genderPrompt)) {
```

File: list2/ex9/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        const salaryPrompt = parseFloat(prompt('>'));
+
+        if (isNaN(salaryPrompt)) {
+            console.log('Insert a number to the salary.');
+            return;
+        }
+
+        if (genderPrompt === 'M') {
+            totalMenSalary += salaryPrompt;
+        } else if (genderPrompt === 'F') {
```
Comment (emiliosheinz) [inline list2/ex9/index.js]:
```suggestion
        if (genderPrompt === GENDER.MALE) {
            totalMenSalary += salaryPrompt;
        } else if (genderPrompt === GENDER.FEMALE) {
```

File: list2/ex9/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            totalFemaleSalary += salaryPrompt;
+        }
+        
+
+        console.log('1. Continue the loop.');
+        console.log('0. Exit.');
+        continueLoop = parseInt(prompt('>'));
+
+    } while (continueLoop === 1);
```
Comment (emiliosheinz) [inline list2/ex9/index.js:42]: Great usage of do while

**Review by emiliosheinz — APPROVED** [review]

File: list2/ex9/index.js (hunk 3 above)
Comment (PedroPapa1) [reply list2/ex9/index.js:42]: Thanks!

**Pull request merged**